composeApp/src/commonMain/kotlin/org/ooni/probe/data/repositories/MeasurementRepository.kt

```diff
@@ -54,6 +60,17 @@ class MeasurementRepository(
             .mapToList(backgroundContext)
             .map { list -> list.mapNotNull { it.toModel() } }
 
+    fun selectTestKeysByResultId(filter: ResultFilter): Flow<List<TestKeysWithResultId>> {
+        val descriptorFilter = (filter.descriptor as? ResultFilter.Type.One)?.value
+        return database.measurementQueries
+            .selectTestKeysByResultId(
```

Comment:
It says byResultId by you are only filtering by descriptor. Are you already preparing for the other task showing the summary inside the result? Nevertheless, the name of the query feels wrong.